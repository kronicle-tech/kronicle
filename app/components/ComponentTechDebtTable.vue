<template>
  <table
    v-if="componentsAndTechDebts && componentsAndTechDebts.length > 0"
    class="table table-dark table-bordered table-striped mt-2"
    style="width: 100%"
  >
    <thead>
      <tr>
        <th>Name</th>
        <th>Tech Debts</th>
      </tr>
    </thead>
    <tbody>
      <tr
        v-for="componentAndTechDebts in componentsAndTechDebts"
        :key="componentAndTechDebts.component.id"
      >
        <td class="component-name table-primary">
          <ComponentName :component="componentAndTechDebts.component" />
        </td>
        <td>
          <TechDebts :tech-debts="componentAndTechDebts.techDebts" />
        </td>
      </tr>
    </tbody>
  </table>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import ComponentName from '~/components/ComponentName.vue'
import TechDebts from '~/components/TechDebts.vue'
import { ComponentAndTechDebts } from '~/types/component-tech-debts'

export default Vue.extend({
  components: {
    ComponentName,
    TechDebts,
  },
  props: {
    componentsAndTechDebts: {
      type: Array as PropType<ComponentAndTechDebts[]>,
      default: undefined,
    },
  },
})
</script>
