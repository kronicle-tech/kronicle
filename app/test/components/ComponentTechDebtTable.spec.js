import { mount } from '@vue/test-utils'
import ComponentTechDebtTable from '@/components/ComponentTechDebtTable.vue'

describe('ComponentTechDebtTable', () => {
  let propsData
  let wrapper
  const createWrapper = () => {
    wrapper = mount(ComponentTechDebtTable, { propsData })
  }

  beforeEach(() => {
    propsData = {}
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when componentsAndTechDebts prop is undefined', () => {
    test('renders nothing', () => {
      createWrapper()
      expect(wrapper.html()).toEqual(``)
    })
  })

  describe('when componentsAndTechDebts is set to multiple components with multiple tech debts', () => {
    beforeEach(() => {
      propsData.componentsAndTechDebts = [
        {
          component: {
            id: 'test-id-1',
            name: 'Test Name 1',
          },
          techDebts: [
            {
              description: 'Test Description 1a',
            },
            {
              description: 'Test Description 1b',
            },
          ],
        },
        {
          component: {
            id: 'test-id-2',
            name: 'Test Name 2',
          },
          techDebts: [
            {
              description: 'Test Description 2a',
            },
            {
              description: 'Test Description 2b',
            },
          ],
        },
      ]
    })

    test('renders the components and their tech debts', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
