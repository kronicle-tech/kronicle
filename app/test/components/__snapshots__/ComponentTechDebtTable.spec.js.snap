// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTechDebtTable when componentsAndTechDebts is set to multiple components with multiple tech debts renders the components and their tech debts 1`] = `
"<table class=\\"table table-dark table-bordered table-striped mt-2\\" style=\\"width: 100%;\\">
  <thead>
    <tr>
      <th>Name</th>
      <th>Tech Debts</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class=\\"component-name table-primary\\"><a href=\\"/components/test-id-1\\">Test Name 1</a></td>
      <td>
        <ul>
          <li><span><span class=\\"badge badge-success\\">Missing Priority</span></span>
            <div>
              <p>Test Description 1a</p>
            </div>
            <div></div>
            <!---->
          </li>
          <li><span><span class=\\"badge badge-success\\">Missing Priority</span></span>
            <div>
              <p>Test Description 1b</p>
            </div>
            <div></div>
            <!---->
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td class=\\"component-name table-primary\\"><a href=\\"/components/test-id-2\\">Test Name 2</a></td>
      <td>
        <ul>
          <li><span><span class=\\"badge badge-success\\">Missing Priority</span></span>
            <div>
              <p>Test Description 2a</p>
            </div>
            <div></div>
            <!---->
          </li>
          <li><span><span class=\\"badge badge-success\\">Missing Priority</span></span>
            <div>
              <p>Test Description 2b</p>
            </div>
            <div></div>
            <!---->
          </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>"
`;
